rerun_js/web-viewer/index.js

```diff
@@ -28,31 +28,93 @@ export class WebViewer {
   /** @type {HTMLCanvasElement | null} */
   #canvas = null;
 
+  /** @type {'ready' | 'starting' | 'stopped'} */
+  #state = "stopped";
+
   /**
    * Start the viewer.
    *
-   * @param {string} [rrd] Optional URL to an `.rrd` file or a WebSocket connection to our SDK.
+   * @param {string | string[]} [rrd] URLs to `.rrd` files or WebSocket connections to our SDK.
    * @param {HTMLElement} [parent] The element to attach the canvas onto.
-   * @returns {Promise<this>}
+   * @returns {Promise<void>}
    */
   async start(rrd, parent = document.body) {
-    if (this.#canvas || this.#handle) return this;
+    if (this.#state !== "stopped") return;
+    this.#state = "starting";
 
-    const canvas = document.createElement("canvas");
-    canvas.id = randomId();
-    parent.append(canvas);
+    this.#canvas = document.createElement("canvas");
+    this.#canvas.id = randomId();
+    parent.append(this.#canvas);
 
     let WebHandle_class = await load();
-    const handle = new WebHandle_class();
-    await handle.start(canvas.id, rrd);
-    if (handle.has_panicked()) {
-      throw new Error(`Web viewer crashed: ${handle.panic_message()}`);
+    if (this.#state !== "starting") return;
+
+    this.#handle = new WebHandle_class();
+    await this.#handle.start(this.#canvas.id, undefined);
+    if (this.#state !== "starting") return;
+
+    if (this.#handle.has_panicked()) {
+      throw new Error(`Web viewer crashed: ${this.#handle.panic_message()}`);
+    }
+
+    if (rrd) {
+      this.open(rrd);
     }
 
-    this.#canvas = canvas;
-    this.#handle = handle;
+    console.log("started");
```

Comment:
If this is an intentional log statement maybe more context?

Reply:
Not intentional, libraries in JS land don't usually log anything, will remove